app/Http/Controllers/AdminDash.php

```diff
@@ -1926,4 +1927,26 @@ public function backgroundMonitor() {
         $format = Config::get('schedule-monitor.date_format');
         return view('dashboard.admin.background-monitor')->with('tasks', $tasks)->with('format', $format);
     }
+
+    public function setLiveEventInfo() {
+        $live_event_info = LiveEvent::find(1);
+        return view('dashboard.admin.live')->with('liveInfo', $live_event_info);
+    }
+
+    public function saveLiveEventInfo(Request $request) {
+        $live_event_info = LiveEvent::find(1);
```

Comment:
I'm not sure how I feel about this depending on something existing in the database like this. I know this pattern exists in the code, but it is quite fragile and I don't think we should be building on it. I don't mind the implementation of a true singleton pattern here, however.

This would look like this:
Let a static function in `LiveEvent` handle pulling the event (i.e. `LiveEvent::getAnnouncement()`)
That function would look something like...
```php
public static function getAnnouncement() {
  $announcement = LiveEvent::first(); // or LiveEvent::find(1);
  if (!$announcement) {
    $announcement = new LiveEvent;
  }

  return $announcement;
}
```

Then anytime we need to pull the announcement, the following will do the trick:
```php
$announcement = LiveEvent::getAnnouncement(); // instead of LiveEvent::find(1);
```

(My syntax in those examples may not be 100% correct, but that's my general idea)

This way we don't depend on the state of the database, and if something weird happens it doesn't require manual intervention. We should also be able to make the constructor for `LiveEvent` private so it won't be created anywhere else.

Reply:
@iccowan I'm working on implementing this. I had something like this originally and then changed it so it more closely mirrored the announcements functionality. What do you mean by making the constructor private? I don't want to override the model constructor here... do I?

Reply:
Implemented your idea here - works well. Great suggestion!

Reply:
> @iccowan I'm working on implementing this. I had something like this originally and then changed it so it more closely mirrored the announcements functionality. What do you mean by making the constructor private? I don't want to override the model constructor here... do I?

Maybe? I can't remember where I did this but I did it at some point. It just prevents from being able to call the constructor so we're forced to use the singleton. However I think that may have been a Java thing in school.

Reply:
@kjporter It looks like we can implement a private constructor, and I think we should here to enforce the singleton pattern. For Laravel, this is what it should look like (to make sure everything gets passed properly):

```php
private function __construct(array $attributes = array()) {
  parent::__construct($attributes);
}
```